src/BSON/Binary.stub.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+  * @generate-class-entries static
+  * @generate-function-entries static
```

Comment:
Also nice to add   `@generate-legacy-arginfo 80000` to take benefit of compatibility stuff when using 8.2. Example when using SensitiveParameter attributes, see

See https://github.com/pierrejoye/php_zip/blob/master/php81/php_zip.stub.php#L91
And https://github.com/pierrejoye/php_zip/blob/master/php81/php_zip_arginfo.h#L532

Reply:
Also notice that is some additional compatibility checks are missing, is still possible to add them in gen_stub.php before 8.2.0 GA

Reply:
If you use 70200 it will also generate a `foo_legady_arginfo.h` without type hinting.

Reply:
`generate-legacy-arginfo 70200` is not supported, so we could only generate legacy arginfo files for PHP 7.0. Since those don't include any argument types at all (which we want since PHP 7.2 allows for this), we'll just do without legacy arginfo for now. We've got everything we wanted covered this way as well:
* argument types (`mixed` and unions only on PHP 8.0+)
* return types for final methods/classes
* tentative return types for non-final methods/classes (PHP 8.1+, no return type info in other versions)

Reply:
generate-legacy-arginfo accepts any PHP_VERISON_ID, "legacy" file is generated as soon as < 80000 ;)

BTW? I really think it should be used, at least for SensitiveParameter.
And hacks for compatibility could probably be proposed in php_src so every ext will be benefit of them.

This is how OpenSource work ;)

Reply:
```
$ ./build/gen_stub.php 
In ./src/MongoDB/WriteConcern.stub.php:
Legacy PHP version must be one of: "70000" (PHP 7.0), "80000" (PHP 8.0), "80100" (PHP 8.1), "80200" (PHP 8.2), "70200" provided
```

I'll take a look at adding a compatibility mode for PHP 7.2, but at the moment it doesn't exist. As it is, legacy arginfo isn't useful for us.